src/DynamoCore/Graph/ModelBase.cs

```diff
@@ -109,6 +112,15 @@ public double X
         }
 
         /// <summary>
+        /// Override in derived classes to specify whether
+        /// to serialize the X property.
+        /// </summary>
+        public virtual bool ShouldSerializeX()
```

Comment:
API Change. The `ShouldSerializeX` and `ShouldSerializeY` methods are added to `NodeModel`.